connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -133,6 +120,40 @@ case class ScalarUserDefinedFunction(
   override def asNondeterministic(): ScalarUserDefinedFunction = copy(deterministic = false)
 }
 
+/**
+ * A special variant of [[ScalarUserDefinedFunction]] for functions that only has one input.
+ */
+private[sql] case class SingleInputUserDefinedFunction[V, K](
```

Comment:
Why we need this? It seems a bit overkill just to deal with a single input column.